Source/WebCore/platform/RectEdges.h

```diff
@@ -173,6 +175,23 @@ inline RectEdges<T>& operator+=(RectEdges<T>& a, const RectEdges<T>& b)
     return a;
 }
 
+template<typename T>
+void RectEdges<T>::unite(const RectEdges<T>& other)
+{
+    // Handle empty special cases first.
+    if (other.isZero())
+        return;
+    if (isZero()) {
+        *this = other;
+        return;
+    }
+
+    setTop(std::max(top(), other.top()));
```

Comment:
This seems to be wrong. Top/left should use `min()`.

Reply:
No I do not think so. The top and left of `RectEdges` are directionless lengths. For example, when a `FloatBoxExtent` is added to a `FloatRect` it expands the `FloatRect`:
```
    void expand(const FloatBoxExtent& box)
    {
        m_location.move(-box.left(), -box.top());
        m_size.expand(box.left() + box.right(), box.top() + box.bottom());
    }
```